stan/math/prim/err/check_symmetric.hpp

```diff
@@ -42,7 +42,7 @@ inline void check_symmetric(
   }
   for (size_type m = 0; m < k; ++m) {
     for (size_type n = m + 1; n < k; ++n) {
-      if (!(fabs(value_of(y(m, n)) - value_of(y(n, m)))
+      if (!(fabs(value_of_rec(y(m, n)) - value_of_rec(y(n, m)))
```

Comment:
Note: This calls value_of_rec once for each entry in the matrix, not twice. y(m, n) runs through the upper triangle and y(n, m) through the lower.